libipl/p10/ipl0.C

```diff
@@ -682,7 +683,14 @@ static int ipl_updatehwmodel(void)
 	    (!fs::exists(BOOTTIME_GUARD_INDICATOR)))
 		apply_fco_override();
 
-	process_guard_records();
+	try {
+		process_guard_records();
+	} catch (const openpower::guard::exception::GuardException &ex) {
+		// capture the exception to use it in the PEL
+		char *exceptionReason = const_cast<char *>(ex.what());
+		ipl_error_callback({IPL_ERR_GUARD_FILE_READ, exceptionReason});
```

Comment:
Why are you throwing an `IPL_ERR_GUARD_FILE_READ` error for any exception from the `openpower::guard::exception::GuardException`? it's not correct, right?

Either we should catch the `GuardFileReadFailed ` exception and throw an `IPL_ERR_GUARD_FILE_READ` error or we should catch the respective exception and throw the respective error from the libipl, right?

https://github.com/open-power/guard/blob/main/libguard/guard_exception.hpp

Reply:
I would like to catch both GuardFileReadFailed  and GuardFileOpenFailed execption in this scenario as the guard file is corrupted. May be I will change the name of IPL_ERR_GUARD_FILE_READ to IPL_ERR_GUARD_FILE_OPERATION. I will discuss with the team on what needs to GuardFileSeekFailed and GuardFileReadFailed